mewarpx/mewarpx/utils_store/profileparser.py

```diff
@@ -0,0 +1,136 @@
+import sys
+import re
+import json
+
+
+class FullProfile(object):
+    """
+    Class for parsing, storing and writing the full profile data
+    """
+    def __init__(self, lines):
+        self.lines = lines
+        self.min = 0
+        self.avg = 0
+        self.max = 0
+        self.profile_dict = {}
+
+    def parse_and_write_full_profiling_output(self):
+        print("Parsing profiling output...")
+        # parse the first line first and then go line by line
+        # first line is of the form: "TinyProfiler total time across processes [min...avg...max]: x ... y ... z"
+        values = self.lines[0].split(":")[1].split("...")
+        self.min = float(values[0].strip())
+        self.avg = float(values[1].strip())
+        self.max = float(values[2].strip())
+
+        self.profile_dict["total_time_across_procs"] = {
+            "min" : self.min,
+            "avg" : self.avg,
+            "max" : self.max
+        }
+
+        self.profile_dict["exclusive_function_profiles"] = {}
+        self.profile_dict["inclusive_function_profiles"] = {}
+
+        # the next 4 lines are not needed for parsing, as is the first line.
+        del self.lines[0:5]
+        index = 0
+        for line in self.lines:
+            if "------" in line:
+                break
+            # these lines are of the form "Function()  ncalls  excl_min  excl_avg   excl_max  max_percent"
+            # with more spaces than are here
+
+            # replace the multiple spaces with a single space
+            line = re.sub("\s+", " ", line)
+            values = line.split(" ")
+
+            name = values[0]
```

Comment:
Warp had some profiling strings that contained spaces in their names. I don't know if WarpX is better/stricter about this.

If it's not, you could probably make this all work by working backwards from the end of the split line, with values coming as

```python
max_percent = foo(values[-1])
excl_max = float(values[-2])
...
name = values[0:-5].join(" ") # check that -5 for correctness
```

Reply:
Even if WarpX is stricter, it's a good idea to be on the safe side.